Branch-SDK/src/main/java/io/branch/referral/PrefHelper.java

```diff
@@ -735,11 +735,13 @@ public JSONObject getReferringURLQueryParameters()  {
 
         String string = getString(KEY_REFERRING_URL_QUERY_PARAMETERS);
         JSONObject params = new JSONObject();
-        try {
-            params = new JSONObject(string);
-        }
-        catch (JSONException e) {
-            PrefHelper.LogException("Unable to get URL query parameters as string: ", e);
+
+        if (string != NO_STRING_VALUE) {
```

Comment:
This is Java so != on strings tests object reference value, not character value. 
Use
`if (!TextUtils.isEmpty(string) && PrefHelper.NO_STRING_VALUE.equals(string)`